storage/range.go

```diff
@@ -159,10 +168,11 @@ type RangeManager interface {
 // integrity by replacing failed replicas, splitting and merging
 // as appropriate.
 type Range struct {
-	Desc      *proto.RangeDescriptor
-	rm        RangeManager // Makes some store methods available
-	stats     *rangeStats  // Range statistics
-	splitting int32        // 1 if a split is underway; updated atomically
+	Desc  *proto.RangeDescriptor
+	rm    RangeManager // Makes some store methods available
+	stats *rangeStats  // Range statistics
+	// 1 if a split, merge, or replica change is underway; updated atomically
+	metaLock int32
```

Comment:
is there another safety layer for this on the level of Raft? What would happen if you trigger a split on the leader, elect a new leader and on it, trigger another split?